test/blackbox/run_blackbox_tests/test_cases_submesh02.c

```diff
@@ -0,0 +1,248 @@
+	int num_nodes = sizeof(nodes) / sizeof(nodes[0]);
+
+	PRINT_TEST_CASE_MSG("Creating Containers\n");
+	destroy_containers();
+	create_containers(nodes, num_nodes);
+
+	return 0;
+}
+
+static int black_box_group0_teardown(void **state) {
+	PRINT_TEST_CASE_MSG("Destroying Containers\n");
+	destroy_containers();
+
+	return 0;
+}
+
+static int black_box_all_nodes_setup(void **state) {
+	const char *nodes[] = { "corenode1" };
+	int num_nodes = sizeof(nodes) / sizeof(nodes[0]);
+
+	PRINT_TEST_CASE_MSG("Creating Containers\n");
+	destroy_containers();
+	create_containers(nodes, num_nodes);
+	PRINT_TEST_CASE_MSG("Created Containers\n");
+	return 0;
+}
+
+static void signal_node_start(node_status_t *node_status, int start,
+                              int end) {
+	int i;
+
+	for(i = start; i <= end; i++) {
+		if(NODE_JOINED != node_status[i].currentEv) {
+			return;
+		}
+	}
+
+	for(i = start; i <= end; i++) {
+		node_step_in_container(node_ids[i], "SIGIO");
+	}
+
+	return;
+}
+
+static bool event_cb(mesh_event_payload_t payload) {
+	static node_status_t node_status[6] = { {INIT_ST, core_node1, 0, 3, false},
+		{INIT_ST, app1_node1, 0, 4, false},
+		{INIT_ST, app2_node1, 0, 4, false},
+		{INIT_ST, core_node2, 0, 4, false},
+		{INIT_ST, app1_node2, 0, 7, false},
+		{INIT_ST, app2_node2, 0, 7, false}
+	};
+
+	fprintf(stderr, "%s(%lu) : ", event_node_name[payload.client_id], time(NULL));
+
+	switch(payload.mesh_event) {
+	case NODE_STARTED:
+		fprintf(stderr, "Node Started\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case NODE_JOINED:
+		fprintf(stderr, "Node Joined\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		signal_node_start(node_status, 1, 5);
+		break;
+
+	case ERR_NETWORK:
+		fprintf(stderr, "Network Error\n");
+		assert(false);
+		break;
+
+	case CHANNEL_OPENED:
+		fprintf(stderr, "Channel Opened\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case CHANNEL_DATA_RECIEVED:
+		fprintf(stderr, "Channel Data Received\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case SIG_ABORT:
+		fprintf(stderr, "SIG_ABORT Received\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+
+	case MESH_EVENT_COMPLETED:
+		fprintf(stderr, "MESH_EVENT_COMPLETED Received\n");
+		assert(change_state(&node_status[payload.client_id], payload.mesh_event));
+		break;
+	}
+
+	if((node_status[0].current_index == node_status[0].max_events)
+	                && (node_status[1].current_index == node_status[1].max_events)
+	                && (node_status[2].current_index == node_status[2].max_events)
+	                && (node_status[3].current_index == node_status[3].max_events)
+	                && (node_status[4].current_index == node_status[4].max_events)
+	                && (node_status[5].current_index == node_status[5].max_events)) {
```

Comment:
Again, if you are doing the same thing repeatedly, a `for`-loop will probably be better. Also, this pattern is repeated multiple times in different files. So why not create a function in `mesh_event_handler.c` that automates this? For example:
```
check_nodes_finished(node_status_t *node_status, int start, int end) {
    for(int i = start; i <= end; i++) {
        if(nodes_status[i].current_index == node_state[i].max_events) {
            return false;
        }
    }

    return true;
}
```
Then you can replace the `if`-statement here with `if(check_nodes_finished(node_status, 0, 5)) {...}`, and similarly in the other places where you do the same thing.